### PR TITLE
fix:  lsp define in nb is not clear when delete vm

### LINKED ISSUE #3424: 版本1.12.0，删除虚拟机后，lsp未正常删除，需要等待gc清理
# Bug Report
删除虚拟机后，lsp未正常删除，需要等待gc清理
## Expected Behavior
删除虚拟机后，lsp应随删除虚拟机pod之后删除。
## Actual Behavior
虚拟机pod后，lsp未正常立即删除，只能等待gc的清理，导致在同vpc下另外一个子网创建同名的虚拟机，lsp重复检测是全局检测，检测重复lsp后并不创建新的lsp。
## Steps to Reproduce the Problem

1. subnet-1 创建虚拟机

2. 暂停、删除虚拟机，等待vm-pod删除，查看lsp依然存在。

3. 在subnet-2创建同名的虚拟机，导致未正常创建lsp

## Additional Info

- Kubernetes version:

  **Output of `kubectl version`: 1.23.15**

### PR DESCRIPTION
**Pull request by watermelon-brother**
## What type of this PR

fix:  lsp define in nb is not clear when delete vm

## Which issue(s) this PR fixes

#3424

## WHAT

If lsp define in nb is not clear, only gc run can clear it.

## HOW

we can think the pod is vm or normal pod when add some logic.

### CODE REVIEW

**Review by bobz965 — APPROVED** [review]

File: pkg/controller/pod.go [hunk 1]
```diff
@@ -992,7 +992,13 @@ func (c *Controller) handleDeletePod(key string) error {
 		}
 	}
 
-	ports, err := c.OVNNbClient.ListNormalLogicalSwitchPorts(true, map[string]string{"pod": key})
+	isVMPod, vmName := isVMPod(pod)
+	podkey := key
+	if isVMPod {
```
Comment (hongzhen-ma) [inline pkg/controller/pod.go:997]: 在设置 keep-vm-ip 参数的情况下，可以使用 vm name 作为 key 查找 lsp
如果没有设置该参数，vm pod 的 lsp name 仍然为 pod name 构成。

因此需要添加  c.config.EnableKeepVMIP 参数判断

**Review by hongzhen-ma — CHANGES_REQUESTED** [review]

File: pkg/controller/pod.go (hunk 1 above)
Comment (watermelon-brother) [reply pkg/controller/pod.go]: ok

Comment (bobz965) [reply pkg/controller/pod.go]: pod ipam release 也需要修改下名字

**Comment by bobz965** [comment]
pr #3476 has fixed this.

**Pull request closed without merging**